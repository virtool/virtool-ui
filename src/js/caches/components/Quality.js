import React from "react";
import { connect } from "react-redux";
import { Quality } from "../../quality/components/Quality";
import { BoxGroup, BoxGroupHeader, BoxGroupSection } from "../../base";

export const CacheQuality = props => (
    <BoxGroup>
        <BoxGroupHeader>Quality</BoxGroupHeader>
        <BoxGroupSection>
            <Quality {...props} />
        </BoxGroupSection>
    </BoxGroup>
);

export const mapStateToProps = state => {
    const { bases, composition, sequences } = state.caches.detail.quality;
    return { bases, composition, sequences };
};

export default connect(mapStateToProps)(CacheQuality);
