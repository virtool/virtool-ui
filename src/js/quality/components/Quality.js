import React from "react";
import styled from "styled-components";
import { getFontSize } from "../../app/theme";
import { useElementSize } from "../../utils/hooks";
import { QualityChart } from "./Chart";
import { drawBasesChart } from "./Bases";
import { drawNucleotidesChart } from "./Nucleotides";
import { drawSequencesChart } from "./Sequences";

const QualityTitle = styled.h5`
    display: flex;
    font-size: ${getFontSize("lg")};
    justify-content: space-between;
`;

export const Quality = ({ bases, composition, sequences }) => {
    const [ref, { width }] = useElementSize();

    return (
        <div ref={ref}>
            {width && (
                <React.Fragment>
                    <QualityTitle>
                        <strong>Quality Distribution at Read Positions</strong>
                    </QualityTitle>
                    <QualityChart createChart={drawBasesChart} data={bases} width={width} />

                    <QualityTitle>
                        <strong>Nucleotide Composition at Read Positions</strong>
                    </QualityTitle>
                    <QualityChart createChart={drawNucleotidesChart} data={composition} width={width} />

                    <QualityTitle>
                        <strong>Read-wise Quality Occurrence</strong>
                    </QualityTitle>
                    <QualityChart createChart={drawSequencesChart} data={sequences} width={width} />
                </React.Fragment>
            )}
        </div>
    );
};
