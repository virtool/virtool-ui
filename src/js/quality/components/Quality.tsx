import React, { RefObject } from "react";
import styled from "styled-components";
import { getFontSize } from "../../app/theme";
import { useElementSize } from "../../utils/hooks";
import { drawBasesChart } from "./Bases";
import { QualityChart } from "./Chart";
import { drawNucleotidesChart } from "./Nucleotides";
import { drawSequencesChart } from "./Sequences";

const QualityTitle = styled.h5`
    display: flex;
    font-size: ${getFontSize("lg")};
    justify-content: space-between;
`;

export function Quality({ bases, composition, sequences }) {
    const [ref, { width }] = useElementSize();

    return (
        <div ref={ref as RefObject<HTMLDivElement>}>
            {width && (
                <>
                    <QualityTitle>
                        <strong>Quality Distribution at Read Positions</strong>
                    </QualityTitle>
                    <QualityChart createChart={drawBasesChart} data={bases} width={width} />

                    <QualityTitle>
                        <strong>Nucleotide Composition at Read Positions</strong>
                    </QualityTitle>
                    <QualityChart createChart={drawNucleotidesChart} data={composition} width={width} />

                    <QualityTitle>
                        <strong>Read-wise Quality Occurrence</strong>
                    </QualityTitle>
                    <QualityChart createChart={drawSequencesChart} data={sequences} width={width} />
                </>
            )}
        </div>
    );
}
