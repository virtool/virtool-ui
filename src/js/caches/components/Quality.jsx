import React from "react";
import { connect } from "react-redux";
import { BoxGroup, BoxGroupHeader, BoxGroupSection } from "../../base";
import { Quality } from "../../quality/components/Quality";

export const CacheQuality = props => (
    <BoxGroup>
        <BoxGroupHeader>Quality</BoxGroupHeader>
        <BoxGroupSection>
            <Quality {...props} />
        </BoxGroupSection>
    </BoxGroup>
);

export const mapStateToProps = state => {
    const { bases, composition, sequences } = state.caches.detail.quality;
    return { bases, composition, sequences };
};

export default connect(mapStateToProps)(CacheQuality);
