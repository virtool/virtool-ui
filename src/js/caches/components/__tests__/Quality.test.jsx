import { shallow } from "enzyme";
import React from "react";
import { describe, expect, it } from "vitest";
import { CacheQuality, mapStateToProps } from "../Quality";

describe("<CacheQuality />", () => {
    it("should render", () => {
        const props = {
            foo: "bar",
        };
        const wrapper = shallow(<CacheQuality {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("mapStateToProps()", () => {
    it("should return props", () => {
        const quality = {
            bases: "foo",
            composition: "bar",
            sequences: "baz",
        };

        const state = {
            caches: {
                detail: {
                    quality,
                },
            },
        };

        const props = mapStateToProps(state);

        expect(props).toEqual(quality);
    });
});
