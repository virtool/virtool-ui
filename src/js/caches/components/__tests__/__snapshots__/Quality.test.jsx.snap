// Vitest Snapshot v1

exports[`<CacheQuality /> > should render 1`] = `
<BoxGroup>
  <BoxGroupHeader>
    Quality
  </BoxGroupHeader>
  <BoxGroupSection>
    <Quality
      foo="bar"
    />
  </BoxGroupSection>
</BoxGroup>
`;
